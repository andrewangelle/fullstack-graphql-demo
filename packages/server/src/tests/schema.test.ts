import 'babel-polyfill'
import {
  makeExecutableSchema,
  mockServer
} from 'graphql-tools';

import {
  typeDefsMock as typeDefs,
  purchaseQuery,
  feedQuery
} from 'tests/mocks'


/** tests */
describe('Queries', () => {
  const mockSchema = makeExecutableSchema({ typeDefs });
  const server = mockServer(mockSchema, {
    Boolean: () => false,
    ID: () => 'af4ae2ad-0057-4d85-bfd5-aacc68326419',
    Int: () => 1,
    Float: () => 12.34,
    String: () => 'A String',
  });

  it('Feed Query', async () => {
    const test = await server.query(feedQuery);
    expect(test).toMatchSnapshot()
  });

  it('Purchases Query', async () => {
    const test = await server.query(purchaseQuery);
    expect(test).toMatchSnapshot()
  });

  it('Viewer Query', async () => {
    const test = await server.query(`query { viewer { me { id } } }`);
    expect(test).toMatchSnapshot()
  })
});

