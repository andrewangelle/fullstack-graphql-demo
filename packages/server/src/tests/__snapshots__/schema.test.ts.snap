// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries Feed Query 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "description": "A String",
        "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
        "name": "A String",
        "price": "A String",
        "varietal": "A String",
        "vendor": Object {
          "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
        },
      },
      Object {
        "description": "A String",
        "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
        "name": "A String",
        "price": "A String",
        "varietal": "A String",
        "vendor": Object {
          "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
        },
      },
    ],
  },
}
`;

exports[`Queries Purchases Query 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "me": Object {
        "purchases": Array [
          Object {
            "amount": "A String",
            "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
            "items": Array [
              Object {
                "description": "A String",
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
              Object {
                "description": "A String",
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
            ],
          },
          Object {
            "amount": "A String",
            "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
            "items": Array [
              Object {
                "description": "A String",
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
              Object {
                "description": "A String",
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
            ],
          },
        ],
      },
    },
  },
}
`;

exports[`Queries Viewer Query 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "me": Object {
        "bizName": "A String",
        "email": "A String",
        "firstName": "A String",
        "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
        "lastName": "A String",
        "permissions": Array [
          "A String",
          "A String",
        ],
        "products": Array [
          Object {
            "description": "A String",
            "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
            "name": "A String",
            "price": "A String",
            "varietal": "A String",
          },
          Object {
            "description": "A String",
            "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
            "name": "A String",
            "price": "A String",
            "varietal": "A String",
          },
        ],
        "purchases": Array [
          Object {
            "amount": "A String",
            "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
            "items": Array [
              Object {
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
              Object {
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
            ],
          },
          Object {
            "amount": "A String",
            "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
            "items": Array [
              Object {
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
              Object {
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
            ],
          },
        ],
        "role": "A String",
        "sales": Array [
          Object {
            "amount": "A String",
            "email": "A String",
            "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
            "items": Array [
              Object {
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
              Object {
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
            ],
          },
          Object {
            "amount": "A String",
            "email": "A String",
            "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
            "items": Array [
              Object {
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
              Object {
                "id": "af4ae2ad-0057-4d85-bfd5-aacc68326419",
                "name": "A String",
                "price": "A String",
              },
            ],
          },
        ],
      },
    },
  },
}
`;
